devices/ICub/src/ICub.cpp

```diff
@@ -0,0 +1,388 @@
+                else
+                {
+                    yError() << LogPrefix << "Failed to open " << pImpl->leftHandFTPort.getName().c_str();
+                    return false;
+                }
+
+                // Connect to wbd right hand ft port
+                std::string rightHandFTPortName = wbdHandsFTSensorsSet.check("rightHand",yarp::os::Value(false)).asString();
+
+                if(pImpl->rightHandFTPort.open("/ICub/rightHantFTSensor:i"))
+                {
+                    if(!yarp::os::Network::connect(rightHandFTPortName,pImpl->rightHandFTPort.getName().c_str()))
+                    {
+                        yError() << LogPrefix << "Failed to connect " << rightHandFTPortName << " and " << pImpl->rightHandFTPort.getName().c_str();
+                        return false;
+                    }
+                    else {
+                        while (!pImpl->rightHandFTPort.getFirstDataFlag()) {
+                            pImpl->rightHandFTPort.useCallback();
+                        }
+                        pImpl->sensorNames.push_back("rightWBDFTSensor");
+                    }
+                }
+                else
+                {
+                    yError() << LogPrefix << "Failed to open " << pImpl->rightHandFTPort.getName().c_str();
+                    return false;
+                }
+
+            }
+
+        }
+
+    }
+
+    std::string ft6dPrefix = getWearableName() + sensor::IForceTorque6DSensor::getPrefix();
+
+    for (size_t s = 0; s < pImpl->sensorNames.size(); ++s) {
+        // Create the new sensors
+         auto ft6d = std::make_shared<ICubImpl::ICubForceTorque6DSensor>(
+            pImpl.get(), ft6dPrefix + pImpl->sensorNames[s]);
+
+         pImpl->ftSensorsMap.emplace(
+                     ft6dPrefix + pImpl->sensorNames[s],
+                     ICubImpl::icubDeviceSensors<ICubImpl::ICubForceTorque6DSensor>{ft6d,s});
+    }
+
+    return true;
+}
+
+// ---------------
+// Generic Methods
+// ---------------
+
+wearable::WearableName ICub::getWearableName() const
+{
+    return pImpl->wearableName;
+}
+
+wearable::WearStatus ICub::getStatus() const
+{
+    wearable::WearStatus status = wearable::WearStatus::Ok;
+
+    for (const auto& s : getAllSensors()) {
+        if (s->getSensorStatus() != sensor::SensorStatus::Ok) {
+            yError() << LogPrefix << "The status of" << s->getSensorName() << "is not Ok ("
+                     << static_cast<int>(s->getSensorStatus()) << ")";
+            status = wearable::WearStatus::Error;
+        }
+    }
+
+    // Default return status is Ok
+    return status;
+}
+
+
+wearable::TimeStamp ICub::getTimeStamp() const
+{
+    // Stamp count should be always zero
+    return {pImpl->timeStamp.time,0};
+}
+
+bool ICub::close()
+{
+    return true;
+}
+
+// =========================
+// IPreciselyTimed interface
+// =========================
+yarp::os::Stamp ICub::getLastInputStamp()
+{
+    // Stamp count should be always zero
```

Comment:
Same here (mutex)

Reply:
I am not certain of it. 
It is not protected by mutex in `XSensSuitDevice` https://github.com/robotology-playground/wearables/blob/039fb255fcdedafc1ff42660a197d8f01ad6cadf/devices/XsensSuit/src/XsensSuit.cpp#L880

Reply:
Same as above, mutex is required